src/zenml/integrations/xgboost/materializers/xgboost_dmatrix_materializer.py

```diff
@@ -47,16 +47,14 @@ def load(self, data_type: Type[Any]) -> xgb.DMatrix:
         filepath = os.path.join(self.uri, DEFAULT_FILENAME)
 
         # Create a temporary folder
-        temp_dir = tempfile.mkdtemp(prefix="zenml-temp-")
-        temp_file = os.path.join(str(temp_dir), DEFAULT_FILENAME)
+        with tempfile.mkdtemp(prefix="zenml-temp-") as temp_dir:
```

Comment:
Looks like this is not a context manager and `TemporaryDirectory` should be used instead.

Reply:
Good catch, resolved.